### ISSUE TITLE
i can't change a clolor of markers or circles?

### ISSUE THREAD

**Issue by 4pmaster** [ellisonbg/ipyleaflet#62]
My enviroment is ubuntu 14.04/anaconda/jupyter notebook.
I've try to do someting with colours of circlemarkers and markers but nothing changes, colour still blue. 
What i can do to fix it? 

Thank you!

**Comment by mxndrwgrdnr** [comment]
+1 for this issue. I'm seeing the same thing.

**Comment by kylehounslow** [comment]
Same for me. Would be nice to have colours not hard-coded as `#0033FF`

**Comment by martinRenou** [comment]
Unfortunately it is not possible to change color of markers in leaflet. But you can change color of a CircleMarker with `my_circle_marker.color = "red"` or `my_circle_marker.color = "#B4D455"`, it works fine, at least on master. 
Keep in mind that when you change the color with the `color` attribute it will only change the border color of the circle, if you change the color with the `fill_color` attribute it will fill the circle with the given color, but you would have to change the `fill_opacity` if you want to see it clearly (`fill_opacity` is equal to 0.2 by default)

**Comment by martinRenou** [comment]
Good point. We should look at what is done in Folium. If you feel like doing it please go ahead @deeplook :) That'd be awesome

**Comment by martinRenou** [comment]
Folium uses https://github.com/lvoogdt/Leaflet.awesome-markers

**Comment by OliverEvans96** [comment]
Awesome icons only seem to support a fixed set of colors, is that true? I'm getting:
```
TraitError: The 'marker_color' trait of an AwesomeIcon instance must be any of ['white', 'red', 'darkred', 'lightred', 'orange', 'beige', 'green', 'darkgreen', 'lightgreen', 'blue', 'darkblue', 'lightblue', 'purple', 'darkpurple', 'pink', 'cadetblue', 'white', 'gray', 'lightgray', 'black'], but a value of '#fde725' <class 'str'> was specified.
```
Have there been any updates on this issue since last year? It would be great to be able to use arbitrary colors to represent a continuous variable at discrete locations with markers + a colorbar.

Thanks,
Oliver

**Comment by martinRenou** [comment]
You are right, you can only use a set of colors, simply because those icons are coming from image files.

We could add a new kind of marker that accept any kind of RGB color as input. It would indeed be a nice feature.

**Comment by giswqs** [comment]
I was also looking for a way to change marker colors. This would be a nice feature to add.

**Comment by martinRenou** [comment]
Currently, the only way to change marker colors is by using custom icons, see https://ipyleaflet.readthedocs.io/en/latest/api_reference/icon.html. You can only use the FontAwesome icons, which allow to change the color: https://ipyleaflet.readthedocs.io/en/latest/api_reference/awesome_icon.html

**Comment by giswqs** [comment]
@martinRenou  Many thanks for the info! Will try it out.

**Comment by deeplook** [comment]
I've tried changing the color of an `AwesomeIcon`, but in vain. Only if I reload the current browser page the new color is shown, weird.

<img width="558" alt="Screenshot 2020-10-15 at 10 42 40" src="https://user-images.githubusercontent.com/1001778/96099424-852ae000-0ed3-11eb-93b4-0514e02b9ff0.png">

**Comment by martinRenou** [comment]
@deeplook https://ipyleaflet.readthedocs.io/en/latest/api_reference/awesome_icon.html#interactions

> And while a CircleMarker can change color it cannot be moved on the map.

Can you provide a code sample that replicate the issue?

**Comment by deeplook** [comment]
There is no `on_move()`:

<img width="676" alt="Screenshot 2020-10-15 at 10 56 20" src="https://user-images.githubusercontent.com/1001778/96100852-19e20d80-0ed5-11eb-86aa-fb7a936b7366.png">

**Comment by martinRenou** [comment]
Indeed. In LeafletJS the CircleMarker **is not** a Marker. I opened an issue for a couple of years ago but I don't think they took it into account.

You can still observe the `location` property though `marker.observe(callback, 'location')`

**Comment by deeplook** [comment]
Yes, but I cannot move the `CircleMarker` with the mouse... ;)

**Comment by martinRenou** [comment]
I guess it's a limitation of LeafletJS, we cannot do anything about it in ipyleaflet
